src/Analyzers/MSTest.Analyzers/ClassInitializeShouldBeValidAnalyzer.cs

```diff
@@ -75,6 +89,11 @@ public override void Initialize(AnalysisContext context)
             return;
         }
 
+        if (namedTypeSymbol.IsGenericType && !isInheritanceModeSet)
```

Comment:
perf: This rule is for generic types only. Moving the first part of this check  namedTypeSymbol.IsGenericType to the start of this method, and returning when type is not generic, would save us allocating all the attributes in GetAttributes() for every method on non-generic type.